### ISSUE TITLE
List of Australian cities is very lacking

### ISSUE THREAD

**Issue by skissane** [vincentmorneau/oracle-geo-data#12]
Your list of Australian cities is very lacking. For example in New South Wales, Australia, you only 28 entries. There are a lot more towns/cities/suburbs/etc that in NSW. A better source of info could be Wikidata, you can use this SPARQL query:

http://tinyurl.com/n66b3p5

That gives 1854 results which seems to me a more plausible number. You can also get similar info for other Australian states by replacing the wd:Q3224 in the query with the relevant Wikidata entry for other Australian states.

I don't know how comprehensive the geographical data in Wikidata for other countries is. I imported a lot of postcode info from Wikipedia into Wikidata for Australia. I could try doing the same for other countries, e.g. Canada, USA, UK, New Zealand, etc, but inevitably it will be a country-by-country thing.

**Comment by skissane** [comment]
Actually this query is better: http://tinyurl.com/lvzrntq

It gives 6675 localities in Australia.

Even that's probably incomplete, but it is much closer to completion than the data you've already got.

You can use the Download link to get CSV/TSV/JSON/etc. You can also use Link > SPARQL endpoint to get a link to a REST endpoint that runs this query.

The problem with using GeoNames as a data source, is it varies greatly in data quality and completeness. This is one example of where the GeoNames data isn't very good.

**Comment by skissane** [comment]
Or maybe you could use a data source like this:

https://github.com/timbennett/australian_postcodes

That's postcodes. Now, it mostly corresponds to the town/city/suburb/locality/etc concept in Australia. It does have a few other things that aren't actually town/city/suburb/locality but which get assigned postcodes in Australia anyway, like universities (look for "UNIVERSITY" in the name, hospitals look for "HOSPITAL", "LAW COURTS,VIC" (a court complex; but everything else with "COURT" in its name is actually a town/suburb), "NORTH POLE" (for children to send letters to Santa), some major buildings (e.g. "PARLIAMENT HOUSE,NSW", "WORLD TRADE CENTRE,VIC", etc.) But it is getting a lot closer.

**Comment by vincentmorneau** [comment]
The goal of this project is to offer a list of the most common populated areas in the world. My initial data source contain over 11 million location names in over 50 categories. I have taken out all cities that contain less than 15 000 habitants, otherwise the list of cities per region would be polluted with un-wanted data. 

You say 6675 **localities** in Australia. Does that mean cities? Wikipedia has only around 45 high population cities (not the best source I admit): https://en.wikipedia.org/wiki/List_of_cities_in_Australia_by_population 

What do you think?

**Comment by skissane** [comment]
I guess the question is what do you plan to use the data for?

For example, in an application, you might ask people to say where they are from, or where their postal address is. In that scenario, a lot of people live in places with less than 15,000 people, so do you want to take away their ability to say where they live?

"You say 6675 localities in Australia. Does that mean cities?" In Australia, there isn't actually any formal definition of a "city". Of course, in informal usage we do talk about "cities", but the term has no precise legal/official meaning. There are a bunch of other terms with precise/legal/formal/official definitions, but none of those other terms cleanly map to the informal concept of "city". (Wikipedia's list of "cities" is semi-arbitrary, it doesn't directly reflect any official source.) A "locality", which is a formally defined concept, basically means any area name used for postal addressing – so some localities are neighbourhoods of major cities and others are isolated rural districts and everything else in between. I think there are more than 6000 localities in Australia, probably closer to 15,000.

By contrast, in the UK, there actually is a formal legal concept of what a "city" is – a place can only call itself a city if the Queen has declared it to be a "city" – but the formal legal concept doesn't match very well to the popular/informal concept. Bournemouth has a population of 180,000; the Bournemouth urban area, extending to a couple of neighbouring towns, has a population of over 450,000 people – but Bournemouth isn't legally a "city", only a "town". By contrast, St Davids in Wales is legally a "city", but it has a population of under 2000 people. In popular usage, Bournemouth is much more of a "city" than St Davids, but legally it is the other way around.

This is why I think the word "city" is a bad choice of term in international contexts. In many parts of the world (e.g. Australia) the term lacks a precise/formal/legal definition, and in other parts of the world (e.g. the UK) the formal legal definition may not be what you want anyway.

Actually, you are in Quebec right? According to what I read in Wikipedia - https://en.wikipedia.org/wiki/Ville#Usage_in_Canada – in Quebec there is actually no formal/legal distinction between cities and towns either. Montreal has 1.8 million people, Barkmere less than 60, but both are legally villes. Anglophones in Quebec observe an informal practice of calling some villes "cities" and others "towns", but that is a purely informal practice with no legal/official basis. This is different from the rest of Canada, where there is a formal/legal/official "city" vs "town" distinction. So Quebec and Australia are actually rather similar in this regard.

**Comment by vincentmorneau** [comment]
I understand that. There is a complexity to find the right terminology here, so thanks for pointing some of the lacking areas.

Can you provide a list of 5-10 localities in Australia that you think should be listed in the "Cities"? That will help me make a better query from the original 11 millions location records.

Some other libraries use the term "populated places" but I find that to be quite vague. I'll think about the terminology some more.

**Comment by tschf** [comment]
Interesting discussion.

If I look here https://en.wikipedia.org/wiki/List_of_cities_in_Australia 

NSW says:

> Since 1993, only local government areas in New South Wales can be declared as "cities" by the Government, under the Local Government Act 1993.

Then it includes a list of cities not matching that criteria. 
..

Take Northern Territory. There are 3 cities in your data:

1. Darwin
2. Palmerston
3. Alice Springs

That gives a lot of uncovered regions - Katherine, Nhulunbuy for starters. Their wikipedia entries say they are towns, even so... :thinking: 

(As an aside/slightly off topic - I'm not sure continent is the correct word in your demo app. Australia is a continent. Maybe "region" is better to use here?)
.. 

tl;dr - If one was using it to select where they live, I'm not sure what one would focus on - esp. in more remote/regional areas.

**Comment by skissane** [comment]
Well, to start with, the national capital (Canberra), the six state capitals (Adelaide, Brisbane, Hobart, Melbourne, Perth, Sydney), and the capital of the Northern Territory (Darwin) – everyone would agree these are cities. Once you get beyond those eight "capital cities", the question of exactly what is and isn't a city, and whether X and Y are two different cities or different parts of the same city, gets a lot more complex and confusing. There isn't necessarily always a single / official legal answer–in some cases government agencies can disagree among themselves on this. For example, the urban area I live in, the Central Coast, with a population of over 300,000 people, is about 1 hour north of Sydney – the state government says it isn't part of Sydney, the federal statistics bureau (ABS) says it is, the federal weather bureau (BOM) says it isn't – and if it isn't part of Sydney, is it a city in its own right? That's actually not an easy question to answer.

When I said there was no legal concept of "city" in Australia, it turns out I am partially wrong. There are 21 places in the state of New South Wales with the legal status of "city" – Albury, Armidale, Bathurst, Blue Mountains, Broken Hill, Campbelltown, Cessnock, Dubbo, Goulburn, Grafton, Lake Macquarie City Centre, Lismore, Lithgow, Liverpool, Orange, Parramatta, Penrith, Sydney, Tamworth, Wagga Wagga, Wollongong. However, there are other confusions – four of those "cities" (Campbelltown, Liverpool, Parramatta, Penrith), are usually thought of as being districts of the city of Sydney instead of separate cities. There are other places commonly considered "cities" which aren't on that list – similar to the UK situation but not as extreme, there are places that are legally only "towns" which actually have greater population than other places that are legally "cities". For example, Port Macquarie is legally only a "town"–although many people will call it a "city" informally–but at 45,000 people it has a greater population than at least 11 of those 21 legal "cities". (In some cases this discrepancy is due to population changes – a town can grow into a city, but its legal status can lag behind the population growth – in some cases a town which has grown into a city may wish to cling to the title of "town" for marketing or sentimental reasons – in other cases cities shrink into shadows of their former selves, but once the status of "city" is legally granted, nobody wants to take it away.)

Many local government areas (LGA) in New South Wales are named "City of X" but a local government area isn't strictly speaking a "city" even if it has that word in its name – an LGA is similar to the concept of a "county" in some other countries – many of them are subdivisons of the Sydney metro area, some are large mostly rural areas that may contain some urban areas within them. A good example of this is the "City of Shoalhaven" – there is an LGA called that but there is actually no city called "Shoalhaven". Shoalhaven is actually a region which is mostly rural but contains some urban centres. The capital and largest urban centre of the "City of Shoalhaven" is Nowra, but legally speaking Nowra is a town not a city. Another confusion is that "Parramatta" is legally a city, and there is also an LGA called "City of Parramatta", but the city "Parramatta" is only one small part of the LGA "City of Parramatta". So, it is complicated. And I am not so sure of the situation in other states, but I can look into it.

The "populated place" terminology I am familiar with from GeoNames. I actually think that terminology is quite good – it means "a place people live" without getting caught up in the terminological quagmire of whether that is a "city" or "town" or "village" or "hamlet" or "suburb" or "neighbourhood" or whatever. Wikipedia prefers the terminology "human settlement" or just "settlement" for short, but also suggests "locality" and "populated place" as synonyms. ("Locality" is somewhat of an Australianism; the word is used in other countries, but I believe it is only in Australia that it is a term officially endorsed by the government–although the Spanish cognate word "localidad" is used with a rather similar meaning in Argentina, Columbia and Mexico.)

**Comment by skissane** [comment]
Another interesting thing I learnt – there used to legally be cities in Quebec, but there are none any more, yet the status of "city" still exists under Quebec law but is disused. C-19, Loi sur les cités et villes / Cities and Towns Act, provides for both cités / cities and villes / towns in Quebec. But, all of the cités / cities have been abolished or converted into villes / towns. The concept of "city" still exists under Quebec law, but it no longer has any instances.

According to Les institutions administratives locales et régionales au Québec by Robert Gravel, p. 12: "Le statut de cité pouvait autrefois être accordé sur demande du conseil de toute municipalité comptant au moins 6000 habitants. Les changements à la loi, survenus en 1968, ne permettent plus cependant une telle distinction dans le statut des municipalités locales. Aujourd'hui le terme cité tend à disparaître et seulement deux municipalités ont encore cette désignation : Côte-Saint-Luc et Dorval. Les cités possèdent exactement les mêmes pouvoirs que les villes." https://books.google.com/books?id=weeCtefhnogC&pg=PA12

So in 1998 there were only two cities (cités) remaining in Quebec, Côte-Saint-Luc and Dorval. Otherwise Quebec only had towns (villes). And today, Côte-Saint-Luc and Dorval are no longer cities, they are towns too. They were both cities until the Montreal merger of 2002, when they became part of the town of Montreal (ville de Montréal) but then with the Montreal demerger in 2006 they were re-established but lost their city status in the re-establishment, so now they are towns.

**Comment by vincentmorneau** [comment]
Thank you. I agree with everything I'll see to make a better implementation in the next release.
